### PR TITLE
Package action and do not checkin node_modules in any branch

### PR DESCRIPTION
**Pull request by bryanmacfarlane**
Checking in node_modules causes many problems even if it's in a distribution branch.  There's no way to ignore just dev dependencies and it's painful to work in master and merge down but keep two different working models between the branches.

ncc has solve this and also works with typescript (even though this template is just js)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,50 +64,46 @@ run()
 
 See the [toolkit documentation](https://github.com/actions/toolkit/blob/master/README.md#packages) for the various packages.
 
-## Publish to a distribution branch
+## Package for distribution
 
-Actions are run from GitHub repos.  We will create a releases branch and only checkin production modules (core in this case). 
+Actions are run from GitHub repos.  Packaging the action will create a self encapsulated distribution runnable from the GitHub graph.
 
 Comment out node_modules in .gitignore and create a releases/v1 branch
```
Comment (JasonEtco) [inline README.md]: Is this line still relevant?

Comment (bryanmacfarlane) [reply README.md]: you are correct

File: README.md [hunk 2]
```diff
@@ -64,50 +64,46 @@ run()
...
 ```bash
-$ npm prune --production
-$ git add node_modules
-$ git commit -a -m "prod dependencies"
 $ git push origin releases/v1
 ```
 
 Your action is now published! :rocket: 
 
 See the [versioning documentation](https://github.com/actions/toolkit/blob/master/docs/action-versioning.md)
```
Comment (bryanmacfarlane) [inline README.md]: I realized this versioning doc needs to get rewritten in light of this.  Will do that next and update this link.

File: package.json [hunk 3]
```diff
@@ -23,9 +24,11 @@
   },
   "homepage": "https://github.com/actions/javascript-action#readme",
   "dependencies": {
-    "@actions/core": "^1.1.1"
+    "@actions/core": "^1.1.1",
+    "dist": "^0.1.2"
```
Comment (joshmgross) [inline package.json]: Was this accidentally installed?

Comment (bryanmacfarlane) [reply package.json]: good catch

File: README.md (hunk 1 above)
Comment (bryanmacfarlane) [reply README.md]: Actually the second sentence added here is still relevant right?  The action is self encapsulated and runs from the dist folder by cloning the repo.

Comment (bryanmacfarlane) [reply README.md]: I updated it

**Comment by bryanmacfarlane** [comment]
thanks @joshmgross 

will iterate with more changes as we get more feedback.

**Comment by thboop** [comment]
We likely want to update the default workflow
https://github.com/actions/javascript-action/blob/master/.github/workflows/test.yml#L7

We point the `releases/*` branches which is no longer a format we follow for releases. It may actually be hard to detect release branches, because v* would detect anything starting with a `v`. Something like v[0-9][0-9] might work.

We almost may want to look into a step in that workflow that ensures there are no unstaged changes after `npm run  package` is run, that prevents a user from making changes to the source files and not generating the corresponding dist changes, or a malicious user trying to sneak something into the dist file when reviewers think it was auto generated.
`git diff-index --quiet HEAD --` should work
https://stackoverflow.com/questions/3878624/how-do-i-programmatically-determine-if-there-are-uncommitted-changes

File: README.md (hunk 1 above)
Comment (thboop) [reply README.md]: I think we need to remove this line:
Comment out node_modules in .gitignore and create a releases/v1 branch

**Comment by bryanmacfarlane** [comment]
https://github.com/actions/javascript-action/issues/7 for @JasonEtco suggestion.  Will make this improvement in next PR

**Comment by bryanmacfarlane** [comment]
@thboop - good catch.  I left that because this just has master but also because I opened #6 ... I got feedback from @jeffmcaffer and I want to close on that and update the toolkit versioning doc.  He had the same instinct about the question I asked today about it being a little nerve wracking to merge and it's live (as opposed to merge, validate and then explicitly go live with tag).  hmmm

File: README.md (hunk 1 above)
Comment (bryanmacfarlane) [reply README.md]: yeah, I did right before I merged.  Thanks

**Pull request merged**